### ISSUE TITLE
ODR-DabMux needs a better FIC scheduler

### ISSUE THREAD

**Issue by nickpiggottwork** [smallscaledab/small-scale-dab#4]
The FIC channel construction in DabMux.cpp is effective but crude, and makes it difficult to add new FICs.

An alternative would be to create a separate DabFICScheduler.cpp class which creates the FIC (at 96 or 128 bytes per frame) which is streamed into the DabMux through a suitable pipe/MQ.

The DabFICScheduler.cpp should respect the relevant minimum repetition rates specified in TR 101 496-1 s5.10, and should allow easy inclusion of any FIC through use of a 'pluggable' module to construct each FIC.

**Comment by mpbraendli** [comment]
Absolutely, there are a lot of benefits to be gained from refactoring the current FIC scheduler.

Do you know how the FIGs are to be mapped to the FIBs ? It is not clear to me what the precise constraints are on this. I don't think you can "mix and match" them however you want.

I also have a AUDI car receiver that sometimes gets stuck on programme selection, which is very likely due to low repetition rate of some FIGs.

**Comment by mpbraendli** [comment]
Work has started on this topic.

**Comment by nickpiggottwork** [comment]
Great. I'll follow that up over on the list.

**Comment by nickpiggottwork** [comment]
Update - @mpbraendli has re-written the scheduler and you'll find it in the ODR Github. https://github.com/Opendigitalradio/ODR-DabMux/tree/next
